src/main/java/io/github/mfvanek/pg/connection/HighAvailabilityPgConnectionImpl.java

```diff
@@ -53,12 +67,46 @@ public Set<PgConnection> getConnectionsToAllHostsInCluster() {
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary));
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary), primaryHostDeterminer);
     }
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
                                                   @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster);
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
+    }
+
+    @Nonnull
+    public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
+                                                  @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster,
+                                                  @Nullable final Integer delaySeconds) {
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        final HighAvailabilityPgConnectionImpl highAvailabilityPgConnection = new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
```

Comment:
`delaySeconds != null` - it looks like we need a default value here and also we need to implement overloaded method with `delaySeconds`

Reply:
Nice advise! If something might be null it would be readable to have default value.